tests/parametric/test_headers_precedence.py

```diff
@@ -786,6 +795,60 @@ def test_headers_precedence_propagationstyle_resolves_conflicting_contexts(self,
         assert span5["trace_id"] == 6
         assert span5.get("span_links") == None
 
+    @missing_feature(context.library == "java", reason="not_implemented yet")
+    @missing_feature(context.library == "ruby", reason="not_implemented yet")
+    @missing_feature(context.library == "cpp", reason="not_implemented yet")
+    @missing_feature(context.library == "dotnet", reason="not_implemented yet")
+    @missing_feature(context.library == "golang", reason="not_implemented yet")
+    @missing_feature(context.library == "nodejs", reason="not_implemented yet")
+    @missing_feature(context.library == "php", reason="not_implemented yet")
+    @pytest.mark.parametrize("library_env", [{"DD_TRACE_PROPAGATION_STYLE": "tracecontext,datadog,b3multi"}])
+    def test_headers_precedence_propagationstyle_resolves_conflicting_contexts_tracecontext_precedence(
+        self, test_agent, test_library
+    ):
+        """
+        Ensure the last parent id tag is set according to the W3C phase 3 spec
+        """
+        with test_library:
+            # Trace ids with the three styles do not match
+            with test_library.start_span(
+                name="trace_ids_do_not_match",
+                http_headers=[
+                    ["traceparent", "00-11111111111111110000000000000002-000000003ade68b1-01"],
+                    ["tracestate", "dd=s:2;p:000000000000000a,foo=1"],
+                    ["x-datadog-parent-id", "10"],
+                    ["x-datadog-trace-id", "2"],
+                    ["x-datadog-tags", "_dd.p.tid=2222222222222222"],
+                    ["x-b3-traceid", "11111111111111110000000000000003"],
+                    ["x-b3-spanid", "a2fb4a1d1a96d312"],
+                    ["x-b3-sampled", "1"],
+                ],
+            ) as s1:
+                pass
+
+        traces = test_agent.wait_for_num_traces(1)
+        assert len(traces) == 1
+        span = find_span_in_traces(traces, s1.trace_id, s1.span_id)
+
+        assert span["name"] == "trace_ids_do_not_match"
+        span_links = retrieve_span_links(span)
+        assert len(span_links) == 2
+        link1 = span_links[0]
+        assert link1["trace_id"] == 2
+        assert link1["span_id"] == 10
+        assert link1["attributes"] == {"reason": "terminated_context", "context_headers": "datadog"}
+        assert link1["flags"] == 1 | TRACECONTEXT_FLAGS_SET
+        assert link1["trace_id_high"] == 2459565876494606882
+        assert link1.get("tracestate") == None
+
+        link2 = span_links[1]
+        assert link2["trace_id"] == 3
+        assert link2["span_id"] == 11744061942159299346
+        assert link2["attributes"] == {"reason": "terminated_context", "context_headers": "b3multi"}
+        assert link2["flags"] == 1 | TRACECONTEXT_FLAGS_SET
```

Comment:
nit: It would be nice if one of the headers set a sampled tag of `0 | TRACECONTEXT_FLAGS_SET`